sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -95,19 +93,37 @@ export function WalletProvider({
 
 	// Memo-ize the context value so we don't trigger un-necessary re-renders from
 	// ancestor components higher in the component tree.
-	const contextValue = useMemo(
-		() => ({ ...walletState, storageAdapter, storageKey, dispatch }),
-		[storageAdapter, storageKey, walletState],
+	const contextValue = useMemo(() => ({ ...walletState, dispatch }), [walletState]);
+	return (
+		<WalletContext.Provider value={contextValue}>
+			{autoConnect ? (
+				<AutoConnectWallet storageAdapter={storageAdapter} storageKey={storageKey}>
+					{children}
+				</AutoConnectWallet>
+			) : (
+				children
+			)}
+		</WalletContext.Provider>
 	);
-	return <WalletContext.Provider value={contextValue}>{children}</WalletContext.Provider>;
 }
 
-export function useWalletContext() {
-	const context = useContext(WalletContext);
-	if (!context) {
-		throw new Error(
-			'Could not find WalletContext. Ensure that you have set up the WalletProvider.',
-		);
-	}
-	return context;
+function sortWallets<AdditionalFeatures extends Wallet['features']>(
```

Comment:
I decided to co-locate this util with the component since there's not really a need for `walletUtils.ts` anymore